tapchannel/allocation.go

```diff
@@ -0,0 +1,526 @@
+	"github.com/lightninglabs/taproot-assets/tapsend"
+	"github.com/lightningnetwork/lnd/input"
+)
+
+var (
+	// ErrMissingInputs is an error that is returned if no inputs were
+	// provided.
+	ErrMissingInputs = fmt.Errorf("no inputs provided")
+
+	// ErrMissingAllocations is an error that is returned if no allocations
+	// were provided.
+	ErrMissingAllocations = fmt.Errorf("no allocations provided")
+
+	// ErrInputOutputSumMismatch is an error that is returned if the sum of
+	// the input asset proofs does not match the sum of the output
+	// allocations.
+	ErrInputOutputSumMismatch = fmt.Errorf("input and output sum mismatch")
+
+	// ErrNormalAssetsOnly is an error that is returned if an allocation
+	// contains an asset that is not a normal asset (e.g. a collectible).
+	ErrNormalAssetsOnly = fmt.Errorf("only normal assets are supported")
+
+	// ErrCommitmentNotSet is an error that is returned if the output
+	// commitment is not set for an allocation.
+	ErrCommitmentNotSet = fmt.Errorf("output commitment not set")
+)
+
+// AllocationType is an enum that defines the different types of asset
+// allocations that can be created.
+type AllocationType uint8
+
+const (
+	// AllocationTypeNoAssets is the default allocation type that is used
+	// when the allocation type is not important or the allocation does not
+	// carry any assets.
+	AllocationTypeNoAssets AllocationType = 0
+
+	// CommitAllocationToLocal is an allocation type that is used for
+	// allocating assets to the local party.
+	CommitAllocationToLocal AllocationType = 1
+
+	// CommitAllocationToRemote is an allocation type that is used for
+	// allocating assets to the remote party.
+	CommitAllocationToRemote AllocationType = 2
+
+	// CommitAllocationHtlcIncoming is an allocation type that is used for
+	// allocating assets to an incoming HTLC output.
+	CommitAllocationHtlcIncoming AllocationType = 3
+
+	// CommitAllocationHtlcOutgoing is an allocation type that is used for
+	// allocating assets to an outgoing HTLC output.
+	CommitAllocationHtlcOutgoing AllocationType = 4
+
+	// SecondLevelHtlcAllocation is an allocation type that is used for
+	// allocating assets to a second level HTLC output (HTLC-success for
+	// HTLCs accepted by the local node, HTLC-timeout for HTLCs offered by
+	// the local node).
+	SecondLevelHtlcAllocation AllocationType = 5
+)
+
+// Allocation is a struct that tracks how many units of assets should be
+// allocated to a specific output of an on-chain transaction. An allocation can
+// be seen as a recipe/instruction to distribute a certain number of asset units
+// to a specific output of an on-chain transaction. The output is mainly
+// identified by its output index but also carries along additional information
+// that is required for making sure the resulting on-chain outputs can be sorted
+// in a deterministic way (that is almost but not exactly following the BIP-69
+// rules for sorting transaction outputs).
+type Allocation struct {
```

Comment:
is this the equivalent of locking utxos on lnd?

after reading godoc looks like it's only about on-chain assets but the allocation types indicate that we're also going to reflect tap channel htlc balance changes via this `Allocation` struct?